SPRINT 10. APRENDIZAJE AUTOMATICO EN NEGOCIOS. 
---

In [36]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

In [37]:
geo_0 = pd.read_csv('dataset/geo_data_0.csv')
geo_0.info()
display(geo_0.head(10))


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   id       100000 non-null  object 
 1   f0       100000 non-null  float64
 2   f1       100000 non-null  float64
 3   f2       100000 non-null  float64
 4   product  100000 non-null  float64
dtypes: float64(4), object(1)
memory usage: 3.8+ MB


id        f0        f1        f2     product
0  txEyH  0.705745 -0.497823  1.221170  105.280062
1  2acmU  1.334711 -0.340164  4.365080   73.037750
2  409Wp  1.022732  0.151990  1.419926   85.265647
3  iJLyR -0.032172  0.139033  2.978566  168.620776
4  Xdl7t  1.988431  0.155413  4.751769  154.036647
5  wX4Hy  0.969570  0.489775 -0.735383   64.741541
6  tL6pL  0.645075  0.530656  1.780266   49.055285
7  BYPU6 -0.400648  0.808337 -5.624670   72.943292
8  j9Oui  0.643105 -0.551583  2.372141  113.356160
9  OLuZU  2.173381  0.563698  9.441852  127.910945

In [18]:
print('Valores nulos ->\n',geo_0.isna().sum())
print('Valores duplicados -> ',geo_0.duplicated().sum())

Valores nulos ->
 id         0
f0         0
f1         0
f2         0
product    0
dtype: int64
Valores duplicados ->  0


In [34]:
features_0 = geo_0.drop(['id','product'],axis = 1)
target_0 = geo_0['product']

features_train_0 , features_valid_0 , target_train_0, target_valid_0 = train_test_split(features_0, target_0, test_size= 0.25, random_state= 54321)

print(features_train_0.shape, features_valid_0.shape)
print(target_train_0.shape, target_valid_0.shape)

(75000, 3) (25000, 3)
(75000,) (25000,)


In [40]:
print('Promedio de los volumenes de reservas en los pozos de petróleo',target_train_0.mean())

Promedio de los volumenes de reservas en los pozos de petróleo 92.47206779423871


In [39]:
model_0 = LinearRegression()
model_0.fit(features_train_0,target_train_0)

predictions_valid_0 = model_0.predict(features_valid_0)
recm_0 = mean_squared_error(target_valid_0 , predictions_valid_0)**0.5

print(recm_0)

37.68341093860808


Tenemos un RECM de 37.68 miles de barriles mal predichos, si lo comparamos con el promedio total de barriles -> 92.47

Podría decirse que se falló casi un 40% en las predicciones. 